src/context/AuthContext.js

```diff
@@ -0,0 +1,59 @@
+import { createContext, useState } from "react";
+import jwt_decode from "jwt-decode";
+import { APIUrl } from "../config/config";
+
+export const AuthContext = createContext({});
+
+export default function AuthContextProvider({ children }) {
+  const [username, setUsername] = useState("abczzz13");
+  const [password, setPassword] = useState("v$*UfieAepz7zuYphJ^^");
+  const [userToken, setUserToken] = useState("");
+  const [decoded, setDecoded] = useState({});
+  const [redirect, setRedirect] = useState(false);
+
+  const login = async (e) => {
+    e.preventDefault();
+    try {
+      const response = await fetch(`${APIUrl}/token`, {
+        method: "POST",
+        headers: {
+          "Content-type": "application/json",
+          Authorization: "Bearer" + userToken && userToken.token,
+        },
+        body: JSON.stringify({
+          username,
+          password,
+        }),
+      });
+
+      const content = await response?.json();
+      setUserToken(content);
+      setRedirect(true);
+      localStorage.setItem("token", content.token);
+      const decoded = jwt_decode(content.token);
+      setDecoded(decoded);
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  const logout = () => {
+    localStorage.clear();
+    setUserToken("");
+    setRedirect(false);
+  };
+
+  const data = {
```

Comment:
userName, passWord, setUserName, setPassword etc.. belongs in the form

just pass down login(userName, password) instead